### PR TITLE
refactor: Remove dfelibs from dependencies

### PR DESCRIPTION
**Pull request by paulgessinger**
This imports the two header files that we use and removes the dependency on dfelibs.

### CODE REVIEW

**Comment by paulgessinger** [comment]
@andiwand @wdconinc

File: Examples/Framework/src/Framework/Sequencer.cpp
Comment (andiwand) [inline Examples/Framework/src/Framework/Sequencer.cpp:1]: less code for the same thing 😄

File: Examples/Io/Csv/src/CsvBFieldWriter.cpp [hunk 1]
```diff
@@ -49,7 +48,7 @@ void CsvBFieldWriter::run(const Config<Coord, Grid>& config,
 
   // Initialize a CSV writer to the specified filename using the specified
   // column names.
-  dfe::io_dsv_impl::DsvWriter<','> writer(fields, config.fileName);
+  Acts::CsvWriter writer(fields, config.fileName);
```
Comment (andiwand) [inline Examples/Io/Csv/src/CsvBFieldWriter.cpp:51]: are header/source files missing? should this live in `Acts::`?

Comment (paulgessinger) [reply Examples/Io/Csv/src/CsvBFieldWriter.cpp:51]: Right I'll move this to the examples namespace.

File: docs/getting_started.md
Comment (andiwand) [inline docs/getting_started.md:1]: are all of these change intended?

**Review by andiwand — APPROVED** [review]

File: docs/getting_started.md
Comment (paulgessinger) [reply docs/getting_started.md:1]: Ah this is my auto-formatter for markdown. I can revert the other changes.

Comment (paulgessinger) [reply docs/getting_started.md:1]: Should be gone now.

**Review by andiwand — DISMISSED** [review]

**Comment by stephenswat** [comment]
Hmm, we actually depend on this in detray, but only for the IO part which we don't need. I'll add a flag to detray to disable the IO build which should rid us of the dfelibs dependency.

**Comment by stephenswat** [comment]
Nevermind, the Acts traccc and detray plugins depend on dfelibs. But I guess these projects can set up their dependencies by themselves.

**Comment by andiwand** [comment]
@paulgessinger there is a clang tidy failure

also there are a couple of sonarcloud suggestions not sure how critical

**Comment by paulgessinger** [comment]
@andiwand clang-tidy fix is in, I think I won't fix the Sonar issues for legacy code like this.

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [d49e2f82564c89b872926dbd05d145c9e433efbf]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7898910)!

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128504)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128503)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128502)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128501)||
|:red_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128500)||
|:red_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128499)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128498)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128497)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128496)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128495)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128494)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128493)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128492)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128491)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128490)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128489)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128488)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128487)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128486)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128485)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128484)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128483)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128482)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128481)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128480)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128479)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128478)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128477)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128476)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128475)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128474)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128473)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42128472)||

**Pull request merged**